lib/apytypes/_array_functions.py

```diff
@@ -229,3 +232,203 @@ def ravel(a):
+    a : :class:`APyFloatArray` or :class:`APyFixedArray`
+        The array whose axes should be reordered.
+    axis1 : int
+      First axis.
+    axis2 : int
+      Second axis.
+
+    Examples
+    --------
+
+    >>> a = APyFloatArray.from_float(
+    ...     [[1.0, 2.0, 3.0], [-4.0, -5.0, -6.0]],
+    ...     exp_bits=5,
+    ...     man_bits=2,
+    ... )
+    >>> from apytypes import APyFloatArray
+    >>> x = APyTypes.from_float([[1, 2, 3]], exp_bits=5, man_bits=2)
+    >>> x.swapaxes(0, 1).to_numpy()
+    array([[1.],
+         [2.],
+         [3.]])
+
+    >>> x = np.from_float([[[0, 1], [2, 3]], [[4, 5], [6, 7]]], exp_bits=5, man_bits=0)
+    >>> x.to_numpy()
+    array([[[0., 1.],
+          [2., 3.]],
+         [[4., 5.],
+          [6., 7.]]])
+
+    >>> x.swapaxes(0, 2).to_numpy()
+    array([[[0., 4.],
+          [2., 6.]],
+         [[1., 5.],
+          [3., 7.]]])
+
+    Returns
+    -------
+    a_swapped :class:`APyFloatArray` or :class:`APyFixedArray`
+        Copy of `a` with axes swapped
+    """
+
+    try:
+        swapaxes = a.swapaxes
+    except AttributeError:
+        raise TypeError(f"Cannot transpose/swapaxis of type: {type(a)}")
+    return swapaxes(axis1, axis2)
+
+
+def expand_dims(a, axis):
+    """
+    Expand the shape of an array.
+
+    Insert a new axis that will appear at the `axis` position in the expanded
+    array shape.
+
+    Parameters
+    ----------
+    a : :class:`APyFloatArray` or :class:`APyFixedArray`
+        The array whose axes should be reordered.
+    axis : int or tuple of ints
+        Position in the expanded axes where the new axis (or axes) is placed.
+
+    Examples
+    --------
+    >>> import apytypes as apy
+    >>> x = apy.ApyFixedArray.from_float([1, 2], int_bits=5, frac_bits=0)
+    >>> x.shape
+    (2,)
+
+    >>> y = apy.expand_dims(x, axis=0)
+    >>> y.to_numpy()
+    array([[1., 2.]])
+    >>> y.shape
+    (1, 2)
+
+    >>> y = apy.expand_dims(x, axis=1)
+    >>> y.to_numpy()
+    array([[1.],
+           [2.]])
+    >>> y.shape
+    (2, 1)
+
+    ``axis`` may also be a tuple:
+
+    >>> y = apy.expand_dims(x, axis=(0, 1))
+    >>> y.to_numpy()
+    array([[[1, 2]]])
+
+    >>> y = apy.expand_dims(x, axis=(2, 0))
+    >>> y.to_numpy
+    array([[[1.],
+            [2.]]])
+
+    Returns
+    -------
+    result : :class:`APyFloatArray` or :class:`APyFixedArray`
+        copy of `a` with the number of dimensions increased.
+    """
+
+    if type(axis) not in (tuple, list):
+        axis = (axis,)
+
+    out_ndim = len(axis) + a.ndim
+    axis = normalize_axis_sequence(axis, out_ndim)
+
+    shape_it = iter(a.shape)
+    shape = tuple([1 if ax in axis else next(shape_it) for ax in range(out_ndim)])
+
+    return a.reshape(shape)
+
+
+# =============================================================================
+# Helpers
+# =============================================================================
+
+
+def normalize_axis(axis: int, ndim: int) -> int:
```

Comment:
I'd make this and the next "private", i.e., start with `_`. (Because this is not something that a user is expected to benefit from?)

Sorry for missing it the first time.